### PR TITLE
Fix obsidian to lava thing

### PR DESCRIPTION
**Pull request by MertUnverdi**

### CODE REVIEW

File: src/main/java/com/iridium/iridiumskyblock/listeners/BucketListener.java [hunk 1]
```diff
@@ -51,48 +50,41 @@ public void onBucketEvent(PlayerBucketEvent event) {
...
-
-        final IslandManager manager = IridiumSkyblock.getInstance().getIslandManager();
-
-        final Optional<Island> islandOptional = manager.getIslandViaLocation(event.getClickedBlock().getLocation());
-        if (!islandOptional.isPresent())
-            return;
-
-        final User user = IridiumSkyblock.getInstance().getUserManager().getUser(event.getPlayer());
+        Player player = event.getPlayer();
+        ItemStack itemInHand = player.getInventory().getItemInMainHand();
```
Comment (dlsf) [inline src/main/java/com/iridium/iridiumskyblock/listeners/BucketListener.java:54]: This won't work in 1.8-1.12

File: src/main/java/com/iridium/iridiumskyblock/listeners/BucketListener.java [hunk 2]
```diff
@@ -51,48 +50,41 @@ public void onBucketEvent(PlayerBucketEvent event) {
...
 
-        if (!manager.getIslandPermission(islandOptional.get(), user, PermissionType.BUCKET))
+        if (!IridiumSkyblock.getInstance().getConfiguration().obsidianBucket
+                || !event.getAction().equals(Action.RIGHT_CLICK_BLOCK)
+                || !event.getClickedBlock().getType().equals(Material.OBSIDIAN)
+                || !itemInHand.getType().equals(Material.BUCKET)) {
             return;
+        }
```
Comment (dlsf) [inline src/main/java/com/iridium/iridiumskyblock/listeners/BucketListener.java:61]:
```suggestion
        if (!(IridiumSkyblock.getInstance().getConfiguration().obsidianBucket
                && event.getAction().equals(Action.RIGHT_CLICK_BLOCK)
                && event.getClickedBlock().getType().equals(Material.OBSIDIAN)
                && itemInHand.getType().equals(Material.BUCKET))) {
            return;
        }
```

File: src/main/java/com/iridium/iridiumskyblock/listeners/BucketListener.java [hunk 3]
```diff
@@ -51,48 +50,41 @@ public void onBucketEvent(PlayerBucketEvent event) {
...
+                }
+            } else {
+                player.sendMessage(StringUtils.color(IridiumSkyblock.getInstance().getMessages().noPermission.replace("%prefix%", IridiumSkyblock.getInstance().getConfiguration().prefix)));
+            }
+        } else {
+            player.sendMessage(StringUtils.color(IridiumSkyblock.getInstance().getMessages().noIsland.replace("%prefix%", IridiumSkyblock.getInstance().getConfiguration().prefix)));
+        }
     }
-
 }
```
Comment (dlsf) [inline src/main/java/com/iridium/iridiumskyblock/listeners/BucketListener.java:90]:
```suggestion
}

```

File: src/main/java/com/iridium/iridiumskyblock/utils/PlayerUtils.java [hunk 4]
```diff
@@ -101,6 +102,16 @@ public static int getTotalExperience(final Player player) {
         return exp;
     }
 
+    /**
+     * check an inventory has free slots
+     *
+     * @param inventory The Inventory for check slots
+     * @return boolean The inventory has empty content
+     */
+    public static boolean hasEmptySlot(Inventory inventory) {
```
Comment (dlsf) [inline src/main/java/com/iridium/iridiumskyblock/utils/PlayerUtils.java:111]: https://github.com/Iridium-Development/IridiumCore/blob/master/plugin/src/main/java/com/iridium/iridiumcore/utils/InventoryUtils.java#L75

**Review by dlsf — CHANGES_REQUESTED** [review]

**Review by dlsf — APPROVED** [review]
If you tested it 🤷🏾

File: src/main/java/com/iridium/iridiumskyblock/listeners/BucketListener.java [hunk 5]
```diff
@@ -51,48 +50,41 @@ public void onBucketEvent(PlayerBucketEvent event) {
...
+        Optional<Island> island = IridiumSkyblock.getInstance().getIslandManager().getIslandViaLocation(location);
+        User user = IridiumSkyblock.getInstance().getUserManager().getUser(player);
 
-        event.setCancelled(true);
+        if (island.isPresent()) {
+            if (IridiumSkyblock.getInstance().getIslandManager().getIslandPermission(island.get(), user, PermissionType.BUCKET)) {
+                event.getClickedBlock().setType(Material.AIR);
+                if (itemInHand.getAmount() > 1) {
+                    itemInHand.setAmount(itemInHand.getAmount() - 1);
+                    Bukkit.getScheduler().runTaskLater(IridiumSkyblock.getInstance(), () -> {
```
Comment (PeachesMLG) [inline src/main/java/com/iridium/iridiumskyblock/listeners/BucketListener.java:72]: I dont get why we are running this a tick later, can somone explain what this lava duplication thing is?

Comment (MertUnverdi) [reply src/main/java/com/iridium/iridiumskyblock/listeners/BucketListener.java:72]: i just edited from Ori's code lol

**Comment by dlsf** [comment]
Did you test it without the task?

**Comment by MertUnverdi** [comment]
> Did you test it without the task?

no cause peaches is dum

**Comment by MertUnverdi** [comment]
fuck wrong repo :( sorry i'll pull back

**Pull request merged**